src/ergw_config.erl

```diff
@@ -268,6 +268,8 @@ validate_option(Opt, Value)
        Opt == proxy_map;
        Opt == path_management ->
     throw({error, {options, {Opt, Value}}});
+validate_option(teid, Value) ->
+    ergw_tei_mngr:validate_options({teid, Value});
```

Comment:
that's not going to work

Reply:
Can you share - why? It's was tested with some fail config data and looks like validation is working as expected.

Reply:
validate_options expects a list or map, you are passing a tuple

Reply:
`ergw_tei_mngr:validate_options/1` expect tuple, and then try put a list in `ergw_config:validate_options/4` and call looks like `ergw_config:validate_options(fun validate_option/1, [{teid, {Len, Prefix}}], [], list).`. So, into `ergw_config:validate_option/4` is set the list.